File: objc/Bridge/Bridge/OpenGLAPI.m
#import "OpenGLAPI.h"

@implementation OpenGLAPI

-(void)drawRectangle:(int)x y:(int)y width:(int)w height:(int)h {
    NSLog(@"Rectangle drawn by OpenGL API");
}

-(void)drawCircle:(int)x y:(int)y radius:(int)r {
    NSLog(@"Circle drawn by OpenGL API");
}

@end

File: swift/Bridge/Bridge/OpenGLAPI.swift
import Foundation

public class OpenGLAPI : GraphicsAPI {
    
    func drawRectangle(_ x: Int, _ y: Int, _ width: Int, _ height: Int) {
        print("Rectangle drawn by OpenGL API")
    }
    
    func drawCircle(_ x: Int, _ y: Int, _ radius: Int) {
        print("Circle drawn by OpenGL API")
    }
}
